### PR TITLE
feat(cart): provide a testing module for mocking facade usage

### PR DESCRIPTION
**Pull request by lderrickable**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: libs/cart/src/facades/cart/cart-facade.interface.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
+import { Observable } from 'rxjs';
+import { Action } from '@ngrx/store';
+
+import { DaffCart } from '../../models/cart';
+import { DaffCartErrors } from '../../reducers/cart-errors.type';
+import { DaffCartErrorType } from '../../reducers/cart-error-type.enum';
+
+export interface DaffCartFacadeInterface<T extends DaffCart = DaffCart> {
```
Comment (griest024) [inline libs/cart/src/facades/cart/cart-facade.interface.ts:8]: extend `DaffStoreFacade<Action>`

File: libs/cart/src/facades/cart/cart.facade.ts [hunk 2]
```diff
@@ -8,11 +8,12 @@ import { DaffCartReducersState } from '../../reducers/public_api';
 import { getDaffCartSelectors } from '../../selectors/public_api';
 import { DaffCartErrors } from '../../reducers/cart-errors.type';
 import { DaffCartErrorType } from '../../reducers/cart-error-type.enum';
+import { DaffCartFacadeInterface } from './cart-facade.interface';
 
 @Injectable({
   providedIn: 'root'
 })
-export class DaffCartFacade<T extends DaffCart = DaffCart> implements DaffStoreFacade<Action> {
+export class DaffCartFacade<T extends DaffCart = DaffCart> implements DaffStoreFacade<Action>, DaffCartFacadeInterface<T> {
```
Comment (griest024) [inline libs/cart/src/facades/cart/cart.facade.ts:16]: don't implement `DaffStoreFacade<Action>` here

File: libs/cart/src/facades/cart/cart-facade.interface.ts (hunk 1 above)
Comment (lderrickable) [reply libs/cart/src/facades/cart/cart-facade.interface.ts]: done

File: libs/cart/src/facades/cart/cart.facade.ts (hunk 2 above)
Comment (lderrickable) [reply libs/cart/src/facades/cart/cart.facade.ts]: removed

**Pull request merged**